### PR TITLE
ACN_CD_InfobloxParserUpdate

### PR DESCRIPTION
**Pull request by chicduong**
Fixes #

## Proposed Changes

  - SyslogMessage was not including the Header (RFC3164) part of the message, only the MSG part, which is the proper way. This update modifies the parser (Kusto Function) to remove the parsing of the Header

### CODE REVIEW

File: Parsers/InfobloxNIOS/InfobloxNIOS.txt [hunk 1]
```diff
@@ -27,21 +27,14 @@
...
-        PID = toint(Parser[4]),
-        subString = tostring(Parser[6])
-    | extend Parser2 = extract_all(@"^(\d{2}\-[a-zA-Z]{3}\-\d{4}\s[0-9\.\:]+)?\s?([a-zA-Z-_]+)(\s|\:)?(.*)", dynamic([1,2,3,4]), subString)
-    | mv-expand todynamic(Parser2)
-    | extend responseTime = todatetime(Parser2[0]),
-            Type = tostring(Parser2[1]),
-            RawData_subString = tostring(Parser2[3])
-    | project-away Parser,Parser2,subString;
+    | extend responseTime = todatetime(Parser[0]),
+            Type = tostring(Parser[1]),
```
Comment (ashwin-patil) [inline Parsers/InfobloxNIOS/InfobloxNIOS.txt:33]: Original Syslog also has Type field in schema which is overlapping with new field Type  from parser

File: Parsers/InfobloxNIOS/InfobloxNIOS.txt [hunk 2]
```diff
@@ -27,21 +27,14 @@
 //
 let RawData = Syslog 
     | where Computer in ("datasource") and Facility == "local6"
-    | extend Parser = extract_all(@"^([a-zA-Z]{3}\s\d{2}\s[0-9\:]+)\s(\S+)\s([a-z_]+)(\[(\d+)\]|\:)(\:\s)?(.*)", dynamic([1,2,3,4,5,6,7]), SyslogMessage)
+    | extend Parser = extract_all(@"^(\d{2}\-[a-zA-Z]{3}\-\d{4}\s[0-9\.\:]+)?\s?([a-zA-Z-_]+)(\s|\:)?(.*)", dynamic([1,2,3,4]), SyslogMessage)
```
Comment (ashwin-patil) [inline Parsers/InfobloxNIOS/InfobloxNIOS.txt:30]: Can you check regex? For me, it is returning Type as May from the derived field from Parser.

Query - Testing on sample data from Github: 

```
let Syslog = (externaldata(Computer:string, Facility:string, SyslogMessage: string, ProcessName:string, Type:string)
[@"https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/Sample%20Data/Syslog/InfobloxNIOS.json"]
with (format="multijson"));
let RawData = Syslog 
    | where Computer in ("datasource") and Facility == "local6"
    | extend Parser = extract_all(@"^(\d{2}\-[a-zA-Z]{3}\-\d{4}\s[0-9\.\:]+)?\s?([a-zA-Z-_]+)(\s|\:)?(.*)", dynamic([1,2,3,4]), SyslogMessage)
    | mv-expand todynamic(Parser)
    | extend responseTime = todatetime(Parser[0]),
            Type = tostring(Parser[1]),
            RawData_subString = tostring(Parser[3])
    | project-away Parser;
RawData
```

File: Parsers/InfobloxNIOS/InfobloxNIOS.txt [hunk 3]
```diff
@@ -191,7 +184,7 @@ let dnszone = RawData
         Message = tostring(dnszone_substring[1])
     | project-away RawData_subString, dnszone_substring, dnszone;
 let dnsclient = RawData
```
Comment (ashwin-patil) [inline Parsers/InfobloxNIOS/InfobloxNIOS.txt:186]: PLease check regex and field parsing

image

**Comment by chicduong** [comment]
@ashwin-patil my apologies, the old sample data was not updated. I have uploaded the updated sample data is reflective of live data stream.

File: Parsers/InfobloxNIOS/InfobloxNIOS.txt (hunk 1 above)
Comment (chicduong) [reply Parsers/InfobloxNIOS/InfobloxNIOS.txt]: I have changed this field in the parser to Log_Type

File: Parsers/InfobloxNIOS/InfobloxNIOS.txt (hunk 2 above)
Comment (chicduong) [reply Parsers/InfobloxNIOS/InfobloxNIOS.txt:30]: I have changed this field in the parser to Log_Type, this data should be reflective in the new sample data

File: Parsers/InfobloxNIOS/InfobloxNIOS.txt (hunk 3 above)
Comment (chicduong) [reply Parsers/InfobloxNIOS/InfobloxNIOS.txt:186]: This should be resolved with the updated data sample

File: Parsers/InfobloxNIOS/InfobloxNIOS.txt [hunk 4]
```diff
@@ -132,7 +125,7 @@ let dhcpsession = RawData
         NetworkRange = tostring(dhcpsession[10])
     | project-away RawData_subString, dhcpsession;
 let dhcpbindupdate = RawData
```
[the remark this answers is not in the document]
Comment (chicduong) [reply Parsers/InfobloxNIOS/InfobloxNIOS.txt:127]: This should be resolved with the updated data sample

File: Parsers/InfobloxNIOS/InfobloxNIOS.txt [hunk 5]
```diff
@@ -112,14 +105,14 @@ let dhcpoffer = RawData
...
     | project-away RawData_subString, dhcpoffer;
 let dhcpexpire = RawData
-    | where LogType == "dhcpd" and Type == "DHCPEXPIRE"
+    | where ProcessName == "dhcpd" and Type == "DHCPEXPIRE"
     | extend dhcpexpire = extract_all(@"\w+\s(\S+)\s\w+\s(\S+)", dynamic([1,2]), RawData_subString)
     | mv-expand todynamic(dhcpexpire)
     | extend Source_IP = tostring(dhcpexpire[0]),  
         Dest_MAC_Address = tostring(dhcpexpire[1])
     | project-away RawData_subString, dhcpexpire;
 let dhcpsession = RawData
```
[the remark this answers is not in the document]
Comment (chicduong) [reply Parsers/InfobloxNIOS/InfobloxNIOS.txt:114]: This is not present in the live data / new sample provided.

File: Parsers/InfobloxNIOS/InfobloxNIOS.txt [hunk 6]
```diff
@@ -74,7 +67,7 @@ let dhcprelease = RawData
         State = tostring(dhcprelease[6])
     | project-away RawData_subString, dhcprelease;
 let dhcpack = RawData
-    | where LogType == "dhcpd" and Type == "DHCPACK"
+    | where ProcessName == "dhcpd" and Type == "DHCPACK"
     | extend dhcpack = extract_all(@"(^\s\w+\s(\S+))\sto\s(\S+)\s\((\S+)\)\svia\s(\S+)(\srelay\s(\S+))?(\slease-duration\s(\d+))?(\s\(([a-zA-Z]+)\))?(\sUID\s(\S+))?", dynamic([1,2,3,4,5,6,7,8,9,10,11]), RawData_subString)
     | mv-expand todynamic(dhcpack)
     | extend IPAddress = tostring(dhcpack[1]),
```
[the remark this answers is not in the document]
Comment (chicduong) [reply Parsers/InfobloxNIOS/InfobloxNIOS.txt:73]: This should be resolved with the updated data sample

File: Parsers/InfobloxNIOS/InfobloxNIOS.txt [hunk 7]
```diff
@@ -27,21 +27,14 @@
 //
 let RawData = Syslog 
     | where Computer in ("datasource") and Facility == "local6"
```
Comment (ashwin-patil) [inline Parsers/InfobloxNIOS/InfobloxNIOS.txt:29]: It seems Facility is set as daemon, if its right in SampleData then we have to change in the query from local6 to daemon.

**Comment by ashwin-patil** [comment]
> @ashwin-patil my apologies, the old sample data was not updated. I have uploaded the updated sample data is reflective of live data stream.

No problem, added couple of things based on new sample data.

File: Parsers/InfobloxNIOS/InfobloxNIOS.txt (hunk 7 above)
Comment (chicduong) [reply Parsers/InfobloxNIOS/InfobloxNIOS.txt:29]: This line is used as an example, I had initially intended it to be commented out, leaving the user to modify it according to how their environment is set up. In the instructions (#2) the user is to change this filter to match their data feed. Let me know if we should change how we present this for the other data connectors as well. @preetikr

Comment (chicduong) [reply Parsers/InfobloxNIOS/InfobloxNIOS.txt:29]: @ashwin-patil that was the only sample we were able to obtain at the moment.

**Comment by preetikr** [comment]
I'll merge this after the respective workbook and analytics changes make it to production.

**Comment by chicduong** [comment]
Following up on this, is there anything outstanding?

**Review by preetikr — APPROVED** [review]

**Comment by preetikr** [comment]
The analytic rule template and workbook changes are now in production so merging the parser change as well.

**Pull request merged**